// @mui-treasury/styles/tabs/apple/appleTabs.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tabItemStyles = exports.tabsStyles = void 0;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var tabsStyles = function tabsStyles() {
  return {
    root: {
      backgroundColor: '#eee',
      borderRadius: 10,
      minHeight: 44
    },
    flexContainer: {
      position: 'relative',
      padding: '0 3px',
      zIndex: 1
    },
    indicator: {
      top: 3,
      bottom: 3,
      right: 3,
      height: 'auto',
      borderRadius: 8,
      backgroundColor: '#fff',
      boxShadow: '0 4px 12px 0 rgba(0,0,0,0.16)'
    }
  };
};

exports.tabsStyles = tabsStyles;

var tabItemStyles = function tabItemStyles(_ref) {
  var palette = _ref.palette,
      breakpoints = _ref.breakpoints;
  return {
    root: _defineProperty({
      '&:hover': {
        opacity: 1
      },
      minHeight: 44,
      minWidth: 96
    }, breakpoints.up('md'), {
      minWidth: 120
    }),
    wrapper: {
      // zIndex: 2,
      // marginTop: spacing(0.5),
      color: palette.text.primary,
      textTransform: 'initial'
    }
  };
};

exports.tabItemStyles = tabItemStyles;
